addons/web/static/src/js/fields/relational_fields.js

```diff
@@ -895,11 +897,33 @@ var FieldX2Many = AbstractField.extend({
         this.lastInitialEvent = undefined;
         if (Object.keys(changes).length) {
             this.lastInitialEvent = ev;
-            this._setValue({
+            // Correctly resets the cursor
```

Comment:
// store the cursor position to restore it once potential onchanges have been applied